cookbooks/finite_strain/finite_strain.cc

```diff
@@ -0,0 +1,114 @@
+#include <aspect/material_model/simple.h>
+#include <aspect/simulator_access.h>
+
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/fe/fe_values.h>
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    using namespace dealii;
+
+    template <int dim>
+    class FiniteStrain : public MaterialModel::Simple<dim>
+    {
+      public:
+
+        virtual void evaluate(const MaterialModel::MaterialModelInputs<dim> &in,
+                              MaterialModel::MaterialModelOutputs<dim> &out) const;
+        virtual void parse_parameters(ParameterHandler &prm);
+    };
+
+  }
+}
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+
+    template <int dim>
+    void
+    FiniteStrain<dim>::
+    evaluate(const MaterialModel::MaterialModelInputs<dim> &in,
+             MaterialModel::MaterialModelOutputs<dim> &out) const
+    {
+      Simple<dim>::evaluate(in, out);
+
+      // we need the velocity gradiens for the finite strain (they are not included in material model inputs)
+      // so we get them from the finite element
```

Comment:
Ah, what a bummer. Creating an `FEValues` object is expensive, but I don't really see a good alternative.